### ISSUE TITLE
Joining a server with password gives wrong error message.

### ISSUE THREAD

**Issue by milochristiansen** [anegostudios/VintageStory-Issues#25]
Joining a 1.5.2.2 server that has a password set results in:
```
22.3.2018 23:31:37 [Error] The connection closed unexpectedly.
22.3.2018 23:31:37 [Error] System.Exception: No bytes were received? O_O
```
Which is not at all what I would expect. The only reason I realized I had forgotten to set the password in my connection setting is because I checked the server log...
```
23:31:06 [Server Notification] A Client attempts connecting via TCP
23:31:06 [Server Event] Milo fails to join (invalid server password).
```

**Comment by tyronx** [comment]
Hm. When I test locally I do get a "Password is invalid" screen after attempting to connect. Did you test with a remote server?

**Comment by milochristiansen** [comment]
Yes, `darkagecraft.net` to be exact.

**Comment by tyronx** [comment]
Ok, added a call to socket.Shutdown(), maybe that helps with the disconnect message arriving properly

**Comment by tyronx** [comment]
`socket.Shutdown()` had no effect, but the problem seems to be confined to this particular disconnect message

**Comment by milochristiansen** [comment]
With client and server both on 1.5.2.6 the related "not whitelisted" issue seems to be fixed (I now properly get the message). I have not tested with a password protected server.

**Comment by tyronx** [comment]
Someone feeling like testing it with a pwd protected server?

**Comment by tyronx** [comment]
Well I could reproduce this again with the darkagecraft server. No idea currently how this bug happens.

**Comment by tyronx** [comment]
Might be fixed in 1.7

**Comment by copygirl** [comment]
Assuming fixed. Please respond if it's not the case.
